persist/fs/persist_manager.go

```diff
@@ -99,6 +161,11 @@ func (pm *persistManager) persist(
 	pm.count++
 	pm.bytesWritten += int64(segment.Len())
 
+	end := pm.nowFn()
+	worked := end.Sub(start) - slept
+	currMetrics.recordWorked(worked)
+	currMetrics.recordSlept(slept)
```

Comment:
Maybe `if slept > 0 { currMetrics.recordSlept(slept) }` to avoid a second `AddInt64` for the usual case?